### PR TITLE
AF-2850: Create project rest api should include project name in the response

### PR DESCRIPTION
**Pull request by akumar074**
* Updated the JobStatus result with correct project name when the project is successfully created by REST

**JIRA**: 
[AF-2850](https://issues.redhat.com/browse/AF-2850)
[RHPAM-3519](https://issues.redhat.com/browse/RHPAM-3519)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by akumar074** [comment]
Jenkins run fdb

**Comment by akumar074** [comment]
Jenkins run fdb

**Comment by akumar074** [comment]
Jenkins run fdb

**Comment by akumar074** [comment]
@barboras7 FDB test failure seems unrelated

File: uberfire-rest/uberfire-rest-backend/src/main/java/org/guvnor/rest/backend/JobRequestHelper.java [hunk 1]
```diff
@@ -186,7 +187,13 @@ public JobResult createProject(final String jobId,
                 return result;
             }
 
+            if (project == null) {
```
Comment (adrielparedes) [inline uberfire-rest/uberfire-rest-backend/src/main/java/org/guvnor/rest/backend/JobRequestHelper.java:190]: I think this part is not covered with a Unit Test, can we add that? Good job!

**Review by adrielparedes — CHANGES_REQUESTED** [review]

**Comment by barboras7** [comment]
Hi @akumar074,

from the Jira description:
If project with same name already available in space then '-1' is appended to project name. There is no way we can identify it from response.  Ideally, REST API should throw exception if project with same name is available or it should include project name in the response of above REST API.

_I tested with the business-central.war from the full downstream build, and this is what I am seeing:_

curl -X POST -u 'testadmin:admin1234;' -H 'Content-type: application/json' 'http://localhost:8080/business-central/rest/spaces/MySpace/projects' --data '{"name": "SampleProject", "groupId": "com.sample", "version": "1.0.0-SNAPSHOT", "description": "SampleProject"}'
{"**jobId":"1621410106629-1"**,"status":"APPROVED","spaceName":"MySpace",**"projectName":"SampleProject"**,"projectGroupId":"com.sample","projectVersion":"1.0.0-SNAPSHOT","description":"SampleProject"} 

$ curl -X POST -u 'testadmin:admin1234;' -H 'Content-type: application/json' 'http://localhost:8080/business-central/rest/spaces/MySpace/projects' --data '{"name": "SampleProject", "groupId": "com.sample", "version": "1.0.0-SNAPSHOT", "description": "SampleProject"}'
**{"jobId":"1621410122779-2"**,"status":"APPROVED","spaceName":"MySpace",**"projectName":"SampleProject"**,"projectGroupId":"com.sample","projectVersion":"1.0.0-SNAPSHOT","description":"SampleProject"}



_My understanding from the Jira description is that the customer would expect something like_

$ curl -X POST -u 'testadmin:admin1234;' -H 'Content-type: application/json' 'http://localhost:8080/business-central/rest/spaces/MySpace/projects' --data '{"name": "SampleProject", "groupId": "com.sample", "version": "1.0.0-SNAPSHOT", "description": "SampleProject"}'
**{"jobId":"1621410122779-2"**,"status":"APPROVED","spaceName":"MySpace",**"projectName":"SampleProject-2"**,"projectGroupId":"com.sample","projectVersion":"1.0.0-SNAPSHOT","description":"SampleProject"}

**Comment by akumar074** [comment]
@barboras7 The create-project REST endpoint schedules a job to create the project. This is an async operation. The response we get doesn't tell about the actual project name created but gives information about the job (Job ID and status). We can see the job status with another endpoint "/jobs/{job-id}". This fix adds the correct project name with a success message in the job status.
We mentioned the same in JIRA comments.

**Review by barboras7 — APPROVED** [review]

**Comment by barboras7** [comment]
Thanks @akumar074, I did not notice the comment. Thank you for explanation, approving the PR.

**Comment by akumar074** [comment]
Jenkins run fdb

**Comment by akumar074** [comment]
@adrielparedes FDB test failures looks unrelated to the change.

**Review by adrielparedes — APPROVED** [review]

**Pull request merged**